File: src/main/java/com/jcloisterzone/feature/CourierLetter.java
package com.jcloisterzone.feature;

import com.jcloisterzone.board.Location;
import com.jcloisterzone.board.Position;
import com.jcloisterzone.board.Rotation;
import com.jcloisterzone.board.pointer.FeaturePointer;
import io.vavr.collection.List;


public class CourierLetter extends TileFeature implements Structure {

    private static final List<FeaturePointer> INITIAL_PLACE = List.of(new FeaturePointer(Position.ZERO, CourierLetter.class, Location.I));

    public CourierLetter() {
        this(INITIAL_PLACE);
    }

    public CourierLetter(List<FeaturePointer> places) {
        super(places);
    }

    @Override
    public Feature placeOnBoard(Position pos, Rotation rot) {
        return new CourierLetter(placeOnBoardPlaces(pos, rot));
    }

    public static String name() {
        return "CourierLetter";
    }
}

File: src/main/ts/com/jcloisterzone/feature/CourierLetter.ts
import { List } from "../../../io/vavr/SeqTypes.js";
import { Location } from "../board/Location.js";
import { Position } from "../board/Position.js";
import type { Rotation } from "../board/Rotation.js";
import { FeaturePointer } from "../board/pointer/FeaturePointer.js";
import type { Feature } from "./Feature.js";
import type { Structure } from "./Structure.js";
import { TileFeature } from "./TileFeature.js";

/** The Courier's letter space (The Courier) — the courier figure is summoned onto it; a meeple
 *  may not be deployed here. */
export class CourierLetter extends TileFeature implements Structure {
  static readonly simpleName = "CourierLetter";

  static readonly INITIAL_PLACE = List.of(new FeaturePointer(Position.ZERO, CourierLetter, Location.I));

  constructor(places: List<FeaturePointer> = CourierLetter.INITIAL_PLACE) {
    super(places);
  }

  placeOnBoard(pos: Position, rot: Rotation): Feature {
    return new CourierLetter(this.placeOnBoardPlaces(pos, rot));
  }
}
